### DISCUSSION TITLE
Are there any AllTalk_TTS API usage examples with Python somewhere?

### DISCUSSION THREAD

**Discussion by ahmetax** [erew123/alltalk_tts#269] [Q&A] [answered]
Hi,
I am using BETA standalone version. Now I want to use it as an API server. I will send a text message to the API and get the generated audio in return. I haven't try it yet. Because, it looks like there are too many configuration details to be performed (model, voice, etc). If there are some practical Python examples, it will be very beneficial for me.
Thanks in advance.

**Comment by erew123** [comment, marked as answer]
Hi @ahmetax 

With v2 of AllTalk, you can send just a line of text OR all the settings. If you send just the text and DONT specify anything else, AllTalk will use all the settings that are set in the Global AllTalk settings page e.g.

![image](https://github.com/erew123/alltalk_tts/assets/35898566/871f5adc-b464-4dfd-8657-2d4ffbd27407)

This is somewhat discussed here in the API Dev section:

![image](https://github.com/erew123/alltalk_tts/assets/35898566/5ebc4928-5d4a-42df-9d7d-08ab5f0d7874)

And here in the Endpoints section:

![image](https://github.com/erew123/alltalk_tts/assets/35898566/735f19ca-0a99-4dea-9974-74786293ff41)

If you specifically want a Python script example of sending things to AllTalk, you can look at `\alltalk_tts\system\TGWUI Extension` and the script.py in there.

Line 286 to 335 is an example of sending either streaming generation OR standard generation (note, many options are NOT available for use with streaming generation).

![image](https://github.com/erew123/alltalk_tts/assets/35898566/57ffeb21-645c-4158-b56c-a06ed31f15a6)

Hope that gives you a bit of insight, though Im not sure what level of detail you may need. 

Thanks

**Reply by ahmetax** [reply]
Thank you very much. I will try it today.

**Comment by fury88** [comment]
SO glad I ran across this build of coqui-TTS.  I'm building an AI project at home and I'm starting to push the coqui-tts build to the limits. Some python programs are crashing trying to load xtts_v2. How much of the coqui-tts is compatible with this here? I'm hoping I don't need to rewrite much. I assume the only thing I'd need to re-write is where the audio is created from the text?

**Reply by erew123** [reply]
@fury88 Its fully compatible with the Coqui TTS models as its using an updated version of the Coqui TTS engine. The API however is not a Coqui thing, but AllTalk specific across all TTS engines within AllTalk. If you are going to use the API you can make it as simple as the text or all the settings if you require. There is more detail in the built in documentation.

**Reply by fury88** [reply]
> @fury88 Its fully compatible with the Coqui TTS models as its using an updated version of the Coqui TTS engine. The API however is not a Coqui thing, but AllTalk specific across all TTS engines within AllTalk. If you are going to use the API you can make it as simple as the text or all the settings if you require. There is more detail in the built in documentation.

Thanks for the reply! Yes, that's why I moved over to this model since it's an enhanced branch! Love it! I was hoping I could just switch out my python to use this with minor tweaks. I'll go through the documentation a bit more to see what I can find. I'm not planning on using the UI version as I have scripts running via cron. I'll start digging! :-)
